aws/resource_aws_api_gateway_stage.go

```diff
@@ -82,6 +162,36 @@ func resourceAwsApiGatewayStageCreate(d *schema.ResourceData, meta interface{})
 		input.CacheClusterSize = aws.String(v.(string))
 		waitForCache = true
 	}
+	if v, ok := d.GetOk("canary_settings"); ok {
+		canarySettings := v.([]interface{})
+		canarySetting, ok := canarySettings[0].(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("At least one field is expected inside canary_settings")
```

Comment:
We do not need to error check the schema here and the other ones below. It can be confusing to see this code and we can rely on a panic during acceptance testing if a future schema change breaks a reference.

* `d.GetOk()` will ensure there's an element in `canarySettings`
* `stage_variable_overrides` will always default to an empty map given the existing schema
* `percent_traffic` will always default to 0.0 given the existing schema
* `use_stage_cache` will always default to false given the existing schema